### ISSUE TITLE
Internal error while creating updates: 'category'

### ISSUE THREAD

**Issue by iansewell** [jprouty/mint-amazon-tagger#124]
Behavior observed: 
  Internal error while creating updates: 'category'
Expected behavior: 

less 2022-12-15_22-49-38.log 

2022-12-15 22:49:38,914 INFO __main__: Running version 1.56
2022-12-15 22:52:15,385 INFO mintamazontagger.mintclient: You may be asked for an auth code at the command line! Be sure to press ENTER after typing the 6 digit code.
2022-12-15 22:52:15,386 INFO __main__: Creating a new webdriver
2022-12-15 22:52:17,421 INFO mintamazontagger.mintclient: Mint Login Flow: Navigating to Mint homepage.
2022-12-15 22:52:18,579 INFO mintamazontagger.mintclient: Mint Login Flow: Clicking "Sign in" button.
2022-12-15 22:52:19,772 INFO mintamazontagger.mintclient: Mint Login Flow: login to be performed by user
2022-12-15 22:52:20,417 INFO mintamazontagger.mintclient: Mint Login Flow: login to be performed by user
2022-12-15 22:52:21,677 INFO mintamazontagger.mintclient: Mint Login Flow: login to be performed by user
2022-12-15 22:52:22,274 INFO mintamazontagger.mintclient: Mint Login Flow: login to be performed by user
2022-12-15 22:52:23,681 INFO mintamazontagger.mintclient: Mint Login Flow: login to be performed by user
2022-12-15 22:52:24,686 INFO mintamazontagger.mintclient: Mint Login Flow: login to be performed by user
2022-12-15 22:52:26,065 INFO mintamazontagger.mintclient: Mint Login Flow: login to be performed by user
2022-12-15 22:52:27,427 INFO mintamazontagger.mintclient: Mint Login Flow: login to be performed by user
2022-12-15 22:52:28,296 INFO mintamazontagger.mintclient: Mint Login Flow: login to be performed by user
2022-12-15 22:52:28,799 INFO mintamazontagger.mintclient: Mint Login Flow: login to be performed by user
2022-12-15 22:52:29,538 INFO mintamazontagger.mintclient: Mint Login Flow: login to be performed by user
2022-12-15 22:52:30,268 INFO mintamazontagger.mintclient: Mint Login Flow: login to be performed by user
2022-12-15 22:52:31,240 INFO mintamazontagger.mintclient: Mint Login Flow: login to be performed by user
2022-12-15 22:52:31,892 INFO mintamazontagger.mintclient: Mint Login Flow: login to be performed by user
2022-12-15 22:52:32,480 INFO mintamazontagger.mintclient: Mint Login Flow: login to be performed by user
2022-12-15 22:52:33,357 INFO mintamazontagger.mintclient: Mint Login Flow: login successful.
2022-12-15 22:52:33,357 INFO mintamazontagger.mintclient: Waiting for Mint Overview
2022-12-15 22:52:40,408 INFO mintamazontagger.mintclient: Mint overview loaded
2022-12-15 22:52:40,409 INFO mintamazontagger.mintclient: Getting Mint categories.
2022-12-15 22:52:41,301 INFO mintamazontagger.mintclient: Getting all Mint transactions since 2021-01-20 to None.
2022-12-15 22:52:43,858 ERROR __main__: Internal error while creating updates: 'category'
Traceback (most recent call last):
  File "mintamazontagger/main.py", line 653, in create_updates
  File "mintamazontagger/main.py", line 709, in do_create_updates
  File "mintamazontagger/tagger.py", line 141, in create_updates
  File "mintamazontagger/mint.py", line 158, in parse_from_json
  File "mintamazontagger/mint.py", line 101, in __init__
  File "mintamazontagger/mint.py", line 60, in pythonify_mint_transaction_dict
KeyError: 'category'
2022-12-15 22:52:43,858 ERROR __main__: Internal error while creating updates: 'category'
2022-12-15 22:53:08,871 INFO __main__: Report Issue Clicked
~

**Comment by danpat** [comment]
Also seeing this with the new 1.57 release (I just grabbed the DMG and manually imported Amazon CSV files).

**Comment by jprouty** [comment]
Please try again with version 1.59 and report your logs here. Please review first and redact any PII before posting. Thanks

**Comment by iansewell** [comment]
@jprouty  It was getting tripped up on a Mortgage payment transaction. I hid that account and it worked. I replaced PII with X's

2023-02-01 11:07:53,162 INFO mintamazontagger.mintclient: Getting all Mint transactions since 2022-08-13 to None.
2023-02-01 11:07:54,189 CRITICAL mintamazontagger.mint: No category for mint transaction: {'type': 'LoanTransaction', 'principal': XX, 'principalCurrency': 'USD', 'interest': XX, 'interestCurrency': 'USD', 'escrow': XX, 'escrowCurrency': 'USD', 'metaData': {'lastUpdatedDate': '2022-11-16T12:21:39Z', 'link': [{'otherAttributes': {}, 'href': '/v1/transactions/XX’, 'rel': 'self'}]}, 'id': ‘XX’, 'accountId': ‘XX’, 'accountRef': {'id': ‘XX’, 'name': ‘XXXXX’, 'type': 'LoanAccount', 'hiddenFromPlanningAndTrends': False}, 'date': datetime.date(2022, 11, 15), 'description': 'PAYMENT', 'amount': XX, 'currency': 'USD', 'status': 'POSTED', 'matchState': 'NOT_MATCHED', 'fiData': {'id': '2022-11-15_1', 'date': '2022-11-15', 'amount': XX, 'description': 'PAYMENT', 'inferredDescription': 'PAYMENT'}, 'isReviewed': False, 'transactionType': 'CREDIT', 'etag': ‘XX’, 'isExpense': False, 'isPending': False, 'discretionaryType': 'NONE', 'isLinkedToRule': False, 'transactionReviewState': 'UNREVIEWED'}
2023-02-01 11:07:54,189 ERROR __main__: Internal error while creating updates: 'category'

**Comment by jprouty** [comment]
Oh interesting. And what category is/was shown in the Mint UI? Possibly "Uncategorized"?

**Comment by jprouty** [comment]
@iansewell, gentle ping on the follow-up question above. Thanks.

**Comment by iansewell** [comment]
Yes, it was uncategorized.  Sorry for the delayed response.

I'm running into a new problem now. with 1.62.   

Not sure you care since we have days left for this app :(

2023-03-03 16:09:23,553 ERROR __main__: Internal error while creating updates: 'inferredCategory'
Traceback (most recent call last):
  File "mintamazontagger/main.py", line 693, in create_updates
  File "mintamazontagger/main.py", line 750, in do_create_updates
  File "mintamazontagger/tagger.py", line 141, in create_updates
  File "mintamazontagger/mint.py", line 162, in parse_from_json
  File "mintamazontagger/mint.py", line 105, in __init__
  File "mintamazontagger/mint.py", line 65, in pythonify_mint_transaction_dict
  File "mintamazontagger/mint.py", line 209, in __init__
  File "mintamazontagger/mint.py", line 60, in pythonify_mint_transaction_dict
KeyError: 'inferredCategory'
2023-03-03 16:09:23,554 ERROR __main__: Internal error while creating updates: 'inferredCategory'
2023-03-03 16:09:25,663 INFO __main__: Report Issue Clicked

I've tried selecting everything in the UI: Do not predict cat, Do not tag cat, I added a category filter, nothing seems to help :(
